pipeline.py

```diff
@@ -69,31 +88,94 @@ class TextProcessingPipeline:
+        for article in articles:
+            raw_text = article.get_raw_text()
+            tokens = self._process(raw_text)
+            cleaned_tokens = []
+            single_tagged_tokens = []
+            for token in tokens:
+                cleaned_tokens.append(token.get_cleaned())
+                single_tagged_tokens.append(token.get_single_tagged())
+            article.save_as(' '.join(cleaned_tokens), 'cleaned')
+            article.save_as(' '.join(single_tagged_tokens), 'single_tagged')
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        cleaned_text = re.sub(r"[^а-яА-Я\s]", "", raw_text)
+        analyzed_cleaned_text = Mystem().analyze(cleaned_text)
+        tokens = []
+        for token in analyzed_cleaned_text:
+            if 'analysis' not in token:
+                continue
+            if not token['analysis']:
+                continue
+            if 'lex' not in token['analysis'][0] or 'gr' not in token['analysis'][0]:
+                continue
+            morphological_token = MorphologicalToken(original_word=token['text'])
+            morphological_token.normalized_form = token['analysis'][0]['lex']
+            morphological_token.tags_mystem = token['analysis'][0]['gr']
+            tokens.append(morphological_token)
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path = Path(path_to_validate)
+    if not path.exists():
+        raise FileNotFoundError
+    if not path.is_dir():
+        raise NotADirectoryError
+    all_ids = []
+    for file in path.iterdir():
+        file_name = file.name
+        full_pattern = re.match(r'\d+_(raw.txt|meta.json|raw.pdf'
+                                r'|cleaned.txt|single_tagged.txt'
+                                r'|multiple_tagged.txt|image.png)', file_name)
+        if not full_pattern:
+            raise InconsistentDatasetError
+        pattern = re.match(r'\d+', file_name)
+        article_id = int(pattern.group(0))
+        if article_id < 1:
+            raise InconsistentDatasetError
+        all_ids.append(article_id)
+    if not all_ids:
+        raise EmptyDirectoryError
+    previous_article_id = 0
+    sorted_all_ids = sorted(all_ids)
+    for article_id in sorted_all_ids:
+        if article_id - previous_article_id > 1:
+            raise InconsistentDatasetError
+        previous_article_id = article_id
+    print(sorted_all_ids)
+    if sorted_all_ids[0] != 1:
+        raise InconsistentDatasetError
+    for number in set(sorted_all_ids):
+        name_for_raw = f'{number}_raw.txt'
+        name_for_meta = f'{number}_meta.json'
+        raw_path = path / name_for_raw
+        meta_path = path / name_for_meta
+        if not raw_path.exists() or not meta_path.exists():
+            raise InconsistentDatasetError
```

Comment:
please enrich each raise statement with a message describing the problem. this message should be passed as an argument during error instantiation.

For example, I want to see a message "There are not meta or raw files for an article ID: 3"